src/auxiliary/t8code.jl

```diff
@@ -254,20 +332,175 @@ function trixi_t8_fill_mesh_info(forest, elements, interfaces, mortars, boundari
+                                                         orientation,
+                                                         local_num_conform)
+                            # Non-conforming interface.
+                        elseif level < neighbor_level
+                            local_num_mortars += 1
+
+                            # Last entry is the large element.
+                            mortars.neighbor_ids[end, local_num_mortars] = current_index + 1
+
+                            init_mortar_neighbor_ids!(mortars, iface, dual_faces[1],
+                                                      orientation, neighbor_ielements,
+                                                      local_num_mortars)
+
+                            init_mortar_node_indices!(mortars, (dual_faces[1], iface),
+                                                      orientation, local_num_mortars)
+
+                            # else: `level > neighbor_level` is skipped since we visit the mortar interface only once.
+                        end
+                    else # MPI interfaces/mortars.
+                        # Conforming MPI interface.
+                        if level == neighbor_level
+                            local_num_mpi_conform += 1
+
+                            neighbor_global_ghost_itree = ghost_global_treeids[findlast(ghost_tree_element_offsets .<=
+                                                                                        neighbor_ielements[1])]
+
+                            neighbor_linear_id = neighbor_global_ghost_itree *
+                                                 max_tree_num_elements +
+                                                 t8_element_get_linear_id(neighbor_scheme,
+                                                                          neighbor_leafs[1],
+                                                                          max_level) #UInt64
+
+                            if current_linear_id < neighbor_linear_id
+                                local_side = 1
+                                smaller_iface = iface
+                                smaller_linear_id = current_linear_id
+                                faces = (iface, dual_faces[1])
+                            else
+                                local_side = 2
+                                smaller_iface = dual_faces[1]
+                                smaller_linear_id = neighbor_linear_id
+                                faces = (dual_faces[1], iface)
+                            end
+
+                            global_interface_id = 2 * ndims(mesh) * smaller_linear_id +
+                                                  smaller_iface
+
+                            mpi_mesh_info.mpi_interfaces.local_neighbor_ids[local_num_mpi_conform] = current_index +
+                                                                                                     1
```

Comment:
```suggestion
                            mpi_mesh_info.mpi_interfaces.local_neighbor_ids[local_num_mpi_conform] = current_index + 1
```
Again not sure if this is valid with formatting